lib/compat/wordpress-6.5/interactivity-api/class-wp-interactivity-api-directives-processor.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * Interactivity API: WP_Interactivity_API_Directives_Processor class.
+ *
+ * @package WordPress
+ * @subpackage Interactivity API
+ */
+
+if ( class_exists( 'WP_Interactivity_API_Directives_Processor' ) ) {
+	return;
+}
+
+/**
+ * Class used to iterate over the tags of an HTML string and help process the
+ * directive attributes.
+ */
+class WP_Interactivity_API_Directives_Processor extends Gutenberg_HTML_Tag_Processor_6_5 {
+	/**
+	 * Returns the content between two balanced tags.
+	 *
```

Comment:
if this _must_ be public in order for the Interactivity API to call it internally then we can leave this public, but we can add ` * @access private` to note that this is an internal method only

Reply:
Done in https://github.com/WordPress/gutenberg/pull/58066/commits/5d68f196f68b8aeaaebfc592f5c5c2fe1f25f540.